src/dataframe/dataframe.jl

```diff
@@ -818,15 +818,12 @@ function select!(df::DataFrame, inds::AbstractVector{Int})
     if !allunique(inds)
         throw(ArgumentError("indices must not contain duplicates"))
     end
-
-    targetnames = _names(df)[inds]
-    for i in setdiff(ncol(df):-1:1, inds)
-        splice!(_columns(df), i)
-        delete!(index(df), i)
+    copy!(_names(df), _names(df)[inds])
+    copy!(_columns(df), _columns(df)[inds])
+    empty!(index(df).lookup)
+    for (i, n) in enumerate(_names(df))
+        index(df).lookup[n] = i
```

Comment:
Store `index(df)` in a variable to avoid calling the function over and over.

Reply:
This is what I initially proposed, but I think the compiler should be able to optimize out this so I thought it was OK. But for clarity maybe even `index(df).lookup` could be stored in a variable to make sure all is optimal here.

Reply:
I verified that the compiler does make this optimization by looking at `@code_typed`. Is it still required ?

Reply:
Maybe yes, just to make sure it won't regress.